go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/Azure/azure-container-networking
 
-go 1.22.0
+go 1.23.1
```

Comment:
Setting the required go version to be above that of the defined toolchain depreciates its usage.

https://go.dev/ref/mod#go-mod-file-toolchain